src/adapter/src/coord/sequencer.rs

```diff
@@ -3912,37 +3891,57 @@ impl Coordinator {
-    /// cluster for the given storage host configuration.
-    fn create_linked_cluster_op(
+    /// cluster for the given storage cluster configuration.
+    fn create_linked_cluster_replica_op(
         &mut self,
         on_cluster_name: String,
-        config: &StorageHostConfig,
+        config: &StorageClusterConfig,
     ) -> Result<Option<catalog::Op>, AdapterError> {
-        let size = match config {
-            StorageHostConfig::Managed { size } => size.clone(),
-            StorageHostConfig::Undefined => {
-                let (size, _alloc) = self.catalog.default_storage_host_size();
-                size
-            }
-            StorageHostConfig::Remote { .. } => return Ok(None),
+        let availability_zone = {
+            let azs = self.catalog.state().availability_zones();
+            let n_replicas_per_az = azs
+                .iter()
+                .map(|az| (az.clone(), 0))
+                .collect::<HashMap<_, _>>();
+            Self::choose_az(&n_replicas_per_az)
         };
-        let azs = self.catalog.state().availability_zones();
-        let n_replicas_per_az = azs
-            .iter()
-            .map(|az| (az.clone(), 0))
-            .collect::<HashMap<_, _>>();
-        let location = self.catalog.concretize_replica_location(
-            SerializedComputeReplicaLocation::Managed {
-                size,
-                availability_zone: Self::choose_az(&n_replicas_per_az),
+        let location = match config {
+            StorageClusterConfig::Managed { size } => SerializedComputeReplicaLocation::Managed {
+                size: size.clone(),
+                availability_zone,
                 az_user_specified: false,
             },
-        )?;
+            StorageClusterConfig::Undefined => {
+                let (size, _alloc) = self.catalog.default_storage_cluster_size();
+                SerializedComputeReplicaLocation::Managed {
+                    size,
+                    availability_zone,
+                    az_user_specified: false,
+                }
+            }
+            StorageClusterConfig::Remote { addr } => {
+                // HACK: linked cluster replicas that are remote use the `addrs`
+                // field to transmit exactly one address.
+                //
+                // TODO(benesch): remove the `REMOTE` option from `CREATE
+                // SOURCE` and `CREATE SINK` in favor of `IN CLUSTER` with a
+                // cluster whose replicas are remote.
+                SerializedComputeReplicaLocation::Remote {
+                    addrs: BTreeSet::from_iter(iter::once(addr.to_string())),
+                    compute_addrs: BTreeSet::new(),
+                    workers: NonZeroUsize::new(1).expect("statically nonzero"),
+                }
+            }
+            StorageClusterConfig::Cluster { .. } => {
+                coord_bail!("IN CLUSTER is not yet supported");
```

Comment:
So can the syntax be used at this time or not? Please do make sure to add me to the PR that will make this syntax usable if this does not happen in this PR.

Reply:
It cannot! It has no effect yet. That's why I didn't put this one on your radar, since there are zero expected behavioral changes from this PR, and the existing tests should be sufficiently protective. Will be sure to CC you on the next one, which actually makes `IN CLUSTER` usable.